ssl/ssl_ciph.c

```diff
@@ -532,7 +532,8 @@ int ssl_cipher_get_evp(SSL_CTX *ctx, const SSL_SESSION *s,
         ctmp.id = s->compress_meth;
         if (ssl_comp_methods != NULL) {
             i = sk_SSL_COMP_find(ssl_comp_methods, &ctmp);
-            *comp = sk_SSL_COMP_value(ssl_comp_methods, i);
+            if (i >= 0)
+                *comp = sk_SSL_COMP_value(ssl_comp_methods, i);
```

Comment:
This almost seems like an error state

Reply:
You mean, the case `i < 0` now excluded by the new guard?
Could well be, but I have no insight in what exactly this function is supposed to do in this (corner) case. If there is anything to fix here, I'd say it goes beyond the scope of this PR.